src/DotLiquid.Tests/TemplateTests.cs

```diff
@@ -184,6 +185,134 @@ public void TestRenderToStream()
+        public void TestRenderNullParameters()
+        {
+            Template template = Template.Parse("{{test}}");
+            RenderParameters parameters = null;
+            Assert.Throws<ArgumentNullException>(() => template.Render(parameters));
+        }
+
+        [Test]
+        public void TestRenderToStreamNullParameters()
+        {
+            Template template = Template.Parse("{{test}}");
+            using (Stream stream = new MemoryStream())
+            {
+                Assert.Throws<ArgumentNullException>(() => template.Render(stream, null));
+            }
+        }
+
+        [Test]
+        public void TestRenderToTextWriterNullParameters()
+        {
+            Template template = Template.Parse("{{test}}");
+            using (TextWriter writer = new StringWriter(CultureInfo.InvariantCulture))
+            {
+                Assert.Throws<ArgumentNullException>(() => template.Render(writer, null));
+            }
+        }
+
+        [Test]
+        public void TestGetTagTypeUnknownTag()
+        {
+            Assert.That(Template.GetTagType("unknown"), Is.Null);
+        }
+
+        private class CustomTagFactory : ITagFactory
+        {
+            public string TagName
+            {
+                get { return "custom"; }
+            }
+
+            public Tag Create()
+            {
+                return new CustomTag();
+            }
+
+            public class CustomTag : Tag
+            {
+                public override void Render(Context context, System.IO.TextWriter result)
+                {
+                    result.WriteLine("I am a custom tag");
+                }
+            }
+        }
+
+        private class CustomRawTagFactory : ITagFactory
+        {
+            public string TagName
+            {
+                get { return "customraw"; }
+            }
+
+            public Tag Create()
+            {
+                return new CustomRawTag();
+            }
+
+            public class CustomRawTag : RawBlock
+            {
+                public override void Render(Context context, System.IO.TextWriter result)
+                {
+                    result.WriteLine("I am a raw custom tag");
+                }
+            }
+        }
+
+        [Test]
+        public void TestIsRawTag()
+        {
+            // Trigger the Liquid static constructor to register default Tags.
+            Assert.That(Liquid.ArgumentSeparator, Is.Not.Null);
+            Template.RegisterTagFactory(new CustomTagFactory());
+            Template.RegisterTagFactory(new CustomRawTagFactory());
+
+            Assert.That(Template.IsRawTag("comment"), Is.True);
+            Assert.That(Template.IsRawTag("raw"), Is.True);
+
+            Assert.That(Template.IsRawTag("if"), Is.False);
+            Assert.That(Template.IsRawTag("include"), Is.False);
+
+            Assert.That(Template.IsRawTag("custom"), Is.False);
+            Assert.That(Template.IsRawTag("customraw"), Is.False);

```

Comment:
Again, we are testing for a quirk it seems.

Reply:
Yes. Yes we are.